beacon_node/network/src/beacon_processor/worker/rpc_methods.rs

```diff
@@ -157,6 +157,23 @@ impl<T: BeaconChainTypes> Worker<T> {
                                 "request_root" => ?root
                             );
                         }
+                        Err(BeaconChainError::BlockHashMissingFromExecutionLayer(_)) => {
+                            debug!(
+                                self.log,
+                                "Failed to fetch execution payload for blocks by root request";
+                                "block_root" => ?root,
+                                "reason" => "execution layer not synced",
+                            );
+                            // send the stream terminator
+                            self.send_error_response(
+                                peer_id,
+                                RPCResponseErrorCode::ResourceUnavailable,
+                                "Execution layer not synced".into(),
+                                request_id,
+                            );
+                            drop(send_on_drop);
+                            return;
```

Comment:
just to get the logs below this, maybe a `break` would be better

Reply:
I returned early because we had a stream termination response in the end. Do we send that on error responses? :thinking:

Reply:
I see your point and we shouldn't send more than one terminator, however the log below has been useful in the past to identify/discard issues. Maybe just move it up too, or add a variable to send the right terminator below?

Reply:
Fixed in [5587e4d](https://github.com/sigp/lighthouse/pull/3365/commits/5587e4d2091ff9a6c31b85df58ee71076bff31a9)